lib/envelope.ts

```diff
@@ -53,23 +47,27 @@ class ClientEnvelope extends Envelope {
     }
 
     async verify() {
-
-        const prefix = new TextDecoder().decode(this.signature.slice(0, 1))
-        let sigType: KeypairType
-        if (prefix == 'e') {
-            sigType = KPType.ed25519
-        } else if (prefix == 's') {
-            sigType = KPType.sr25519
-        } else {
-            return false;
+        try {
+            const prefix = new TextDecoder().decode(this.signature.slice(0, 1))
+            let sigType: KeypairType
+            if (prefix == 'e') {
+                sigType = KPType.ed25519
+            } else if (prefix == 's') {
+                sigType = KPType.sr25519
+            } else {
+                return false;
+            }
+            // get twin of sender from twinid
+            this.twin = await getTwinFromTwinID(this.source.twin, this.chainUrl)
+            // get sender pk from twin , update signer to be of sender 
+            await this.getSigner(sigType);
+            // verify signature using challenge and pk
+            const dataHashed = new Uint8Array(this.challenge());
+            return this.signer.verify(dataHashed, this.signature.slice(1), this.signer.publicKey);
+
+        } catch (err) {
+            console.log('invalid response', err)
```

Comment:
I think the code in the try block won't fail except if we couldn't get the source twin from the chain so I think this message is a bit confusing